### PR TITLE
fix: Fixing Analyzer Rules and Updating Values

### PR DESCRIPTION
**Pull request by xoscar**
This PR updates the analyzer rules to better match the semantic conventions.

## Changes

- Fixes issues with rules reporting the results
- Updates required values for different span types
- Updates name matching for Redis db spans
- Updates pokeshop proto file to include the `isFixed` flag

## Fixes

- #2714 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: server/linter/plugins/security/rules/enforce_secure_protocol.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ func NewEnforceHttpsProtocolRule() model.Rule {
 			Name:        "Enforce HTTPS Protocol",
 			Description: "Ensure all request use https",
 			Tips:        []string{},
-			Weight:      40,
+			Weight:      20,
```
Comment (xoscar) [inline server/linter/plugins/security/rules/enforce_secure_protocol.go:25]: Updating the weight

File: server/linter/plugins/standards/rules/ensure_span_naming_rule.go [hunk 2]
```diff
@@ -24,13 +25,18 @@ func NewEnsureSpanNamingRule() model.Rule {
 
 func (r ensureSpanNamingRule) Evaluate(ctx context.Context, trace model.Trace) (model.RuleResult, error) {
 	results := make([]model.Result, 0)
+	hasErrors := false
 	for _, span := range trace.Flat {
-		results = append(results, r.validateSpanName(ctx, span))
+		result := r.validateSpanName(ctx, span)
```
Comment (xoscar) [inline server/linter/plugins/standards/rules/ensure_span_naming_rule.go:30]: We had an issue here that we never reported 100% pass for this rule

File: server/linter/plugins/standards/rules/ensure_span_naming_rule.go [hunk 3]
```diff
@@ -77,18 +83,22 @@ func (r ensureSpanNamingRule) validateHTTPSpanName(ctx context.Context, span *mo
...
+	dbName := span.Attributes.Get("db.name")
 	tableName := span.Attributes.Get("db.sql.table")
+	expectedName := fmt.Sprintf("%s %s.%s", dbOperation, dbName, tableName)
 
-	if tableName != "" {
-		expectedName = fmt.Sprintf("%s %s %s", dbOperation, dbName, tableName)
-	} else {
-		expectedName = fmt.Sprintf("%s %s", dbOperation, dbName)
+	// TODO: fix this by adding proper validation for all other db systems
+	dbSystem := span.Attributes.Get("db.system")
```
Comment (xoscar) [inline server/linter/plugins/standards/rules/ensure_span_naming_rule.go:92]: For Redis, we have to match the prefix of the statement to validate the name. This should be improved in the future to run checks by db system

File: server/linter/plugins/standards/rules/required_attributes.go [hunk 4]
```diff
@@ -12,7 +12,7 @@ var (
 	httpAttrServer = []string{"http.target", "http.scheme", "net.host.name"}
 
 	databaseAttr   = []string{"db.system"}
-	rpcAttr        = []string{"rpc.system", "neet.peer.name"}
```
Comment (xoscar) [inline server/linter/plugins/standards/rules/required_attributes.go:15]: NEEET

File: web/src/assets/pokeshop.proto.json [hunk 5]
```diff
@@ -1,3 +1,3 @@
...
+  "proto": "syntax = \"proto3\";\n\noption java_multiple_files = true;\noption java_outer_classname = \"PokeshopProto\";\noption objc_class_prefix = \"PKS\";\n\npackage pokeshop;\n\nservice Pokeshop {\n  rpc getPokemonList (GetPokemonRequest) returns (GetPokemonListResponse) {}\n  rpc createPokemon (Pokemon) returns (Pokemon) {}\n  rpc importPokemon (ImportPokemonRequest) returns (ImportPokemonRequest) {}\n}\n\nmessage ImportPokemonRequest {\n  int32 id = 1;\n  optional bool isFixed = 2;\n}\n\nmessage GetPokemonRequest {\n  optional int32 skip = 1;\n  optional int32 take = 2;\n  optional bool isFixed = 3;\n}\n\nmessage GetPokemonListResponse {\n  repeated Pokemon items = 1;\n  int32 totalCount = 2;\n}\n\nmessage Pokemon {\n  optional int32 id = 1;\n  string name = 2;\n  string type = 3;\n  bool isFeatured = 4;\n  optional string imageUrl = 5;\n}"
```
Comment (xoscar) [inline web/src/assets/pokeshop.proto.json:2]: Adding support for the `isFixed` flag

Comment (xoscar) [reply web/src/assets/pokeshop.proto.json:2]: In the near future, we should change this to read it directly from the repo to have to do this copy paste

**Review by jorgeepc — APPROVED** [review]

**Pull request merged**